strapi/strapi
# [Rich Text Editor] Save to database

### What does it do?

- Updates the frontend to send the JSON
- Updates the backed to accept type blocks
- Adds the validation to check the schema before saving to the database


### What it does NOT do
- Handle frontend validation (besides a json check). We need to discuss what we should do here and handle it in another PR.
- Handle the validations set by the user. We don't actually have a decision from product what these will be, the usual ones don't really make sense (required, min, max, regex, etc...)

### Why is it needed?

- To persist the content in the RTE

### How to test it?

- Go to the RTE Blocks editor. The mock data has been removed. You can add it back to test everything or test things one by one
- As long as everything you add to the editor state matches the schema we defined in the RFC it should save to the DB
- If you add something to the state that doesn't match the schema ie [{text: ''}], the server should throw a validation error.
- Check the yup schema matches the schema we defined in the RFC


### Unanswered questions
- Should we be sending the shape with version from the frontend {version: '', blocks: []}, or just the array and then reshaping on the server?
- How should we handle frontend validations? 
- What validations will we give to the user (required, min, max), do any of them make sense? You can't have an empty slate editor 🤔 . As mentioned above I think it's another PR
